tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,206 @@
+assert = chai.assert
+
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 42, 42,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
+
+describe 'Annotator.Plugin.EnhancedAnchoring', ->
+  sandbox = null
+  annotator = null
+  data = null
+  ann1 = null
+  ann2 = null
+  ann3 = null
+  target11 = null
+  target12 = null
+  target2 = null
+  target3 = null
+  anchor11 = null
+  anchor12 = null
+  anchor2 = null
+  anchor3 = null
+
+  setupManager = ->
+    annotator =
+      publish: (event) ->
+
+    am = new Annotator.Plugin.EnhancedAnchoring()
```

Comment:
On 2015-02-05 17:36, Aron Carroll wrote:
> Yup, that just begs the question, why is the plugin called
> |EnhancedAnchoring|?

Because Annotator upstream already has a system for dealing with anchors.
With this plugin, we are replacing it with a more sophisticated system.

We could call in EnhancedAnchoringManager.

Imho, simply calling it AnchoringManager would be misleading, because
it would make it sound like this plugin is strictly necessary for anchoring
 - but it is not.

Or at least it used to be an optional addon....
... but given the fact that we have most likely already butchered the
clean separation making this possible, we might as well rename it to
AnchoringManager.

OK, sold.

But let's do it in a separate PR, please.